tools/runner/logger.go

```diff
@@ -0,0 +1,98 @@
+package runner
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	grpcv1 "github.com/grpc/test-infra/api/v1"
+	"github.com/grpc/test-infra/status"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// PodLogger provides functionality to save pod logs to files.
+type PodLogger struct {
+}
+
+// NewPodLogger creates a new PodLogger object.
+func NewPodLogger() *PodLogger {
+	return &PodLogger{}
+}
+
+// savePodLogs saves pod logs to files with same name as pod.
+func (pl *PodLogger) savePodLogs(ctx context.Context, loadTest *grpcv1.LoadTest, podLogDir string) error {
+	// Get pods for this test
+	pods, err := pl.getTestPods(ctx, loadTest)
+	if err != nil {
+		return err
+	}
+
+	// Attempt to create directory. Will not error if directory already exists
+	err = os.MkdirAll(podLogDir, os.ModePerm)
+	if err != nil {
+		return fmt.Errorf("Failed to create pod log output directory %s: %v", podLogDir, err)
+	}
+
+	// Write logs to files
+	errorOccured := false
+	for _, pod := range pods {
+		logFilePath := filepath.Join(podLogDir, pod.Name+".log")
+		err = pl.writePodLogToFile(ctx, pod, logFilePath)
+		if err != nil {
+			errorOccured = true
+		}
+	}
+	if errorOccured {
+		return errors.New("One or more log files could not be written")
+	}
+	return nil
+}
+
+// getTestPods retrieves the pods associated with a LoadTest.
+func (pl *PodLogger) getTestPods(ctx context.Context, loadTest *grpcv1.LoadTest) ([]*corev1.Pod, error) {
+	clientset := getGenericClientset()
```

Comment:
getGenericClientSet is really NewGenericClientSet, so I would make the clientset part of the PodLogger/LogSaver structure.

I would also check if the grpcClientSet can be used to get CoreV1().Pods(...). We probably need only NamespaceDefault in any case (our loadtest pods are always in default namespace).

Reply:
The GRPCTestClientSet as it was could not be used to get Pods (as far as I can tell). I modified the interface slightly to provide this functionality. Do you think the solution works? Relevant code in `clientset/client.go`.  

This would make the function `NewGenericClientset`/`getGenericClientset` unnecessary.